plugins/scaffolder-backend-module-sentry/src/actions/createProject.test.ts

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ConfigReader } from '@backstage/config';
+import { JsonObject } from '@backstage/types';
+import { createSentryCreateProjectAction } from './createProject';
+import { ActionContext } from '@backstage/plugin-scaffolder-node';
+import { InputError } from '@backstage/errors';
+import { randomBytes } from 'crypto';
+
+describe('sentry:project:create action', () => {
+  const createScaffolderConfig = (configData: JsonObject = {}) => ({
+    config: new ConfigReader({
+      scaffolder: {
+        ...configData,
+      },
+    }),
+  });
+
+  const mockFetch = (response = {}) => {
+    const mockedResponse = {
+      status: 201,
+      headers: {
+        get: () => 'application/json',
+      },
+      json: async () =>
+        Promise.resolve({
+          detail: 'project creation mocked result',
+        }),
+      text: async () => Promise.resolve('Unexpected error.'),
+      ...response,
+    };
+    global.fetch = jest
```

Comment:
Not fond of mocking globals like this if we don't have to. Two options:

- Use `msw` to capture all network traffic and mock responses to it. That's a very popular method; check the repo for a ton of usage examples
- Change `createSentryCreateProjectAction` to accept an optional `fetch?: FetchApi` parameter (like [here](https://github.com/backstage/backstage/blob/2d39c1dbdf065c588464c6c0c4fc44bb469aeca0/plugins/graphiql/src/lib/api/GraphQLEndpoints.ts#L55)), and pass in a fake one in each test below

Again, msw is popular and extremely nice because it also ensures that there's absolutely no possibility of accidentally leaking network calls from the tests.

Reply:
> - Use msw to capture all network traffic and mock responses to it. That's a very popular method; check the repo for a ton of usage examples

Agree.  I was reading about fetch calls not being caught by msw and the action uses node global fetch for creating the requests to Sentry. I will try to approach with msw. 

> - Change createSentryCreateProjectAction to accept an optional fetch?: FetchApi parameter (like [here](https://github.com/backstage/backstage/blob/2d39c1dbdf065c588464c6c0c4fc44bb469aeca0/plugins/graphiql/src/lib/api/GraphQLEndpoints.ts#L55)), and pass in a fake one in each test below

Agree. Will do.

Reply:
@freben Updated with the parameter added.